modules/terraform/options.go

```diff
@@ -37,3 +40,54 @@ type Options struct {
 	Logger                   *logger.Logger         // Set a non-default logger that should be used. See the logger package for more info.
 	Parallelism              int                    // Set the parallelism setting for Terraform
 }
+
+// Clone makes a deep copy of the Options object and returns it.
+func (options *Options) Clone() (*Options, error) {
+	newOptions := Options{}
+	if err := copier.Copy(&newOptions, options); err != nil {
```

Comment:
Oh, neat. We should prob use this in Terragrunt in the future too!

Reply:
This library has a few issues with nested structs, so it probably will be problematic in Terragrunt. I added some comments on which fields will not be a deep clone: 2b049ca

Reply:
Noted

Reply:
Actually, wait... If we don't deep copy those two fields, and perhaps other nested structs in the future, this may cause issues. For example, someone may be doing table-driven tests of a single Terraform module, where they test the module in 20 different configurations using `t.Run()` and `t.Parallel()` to create sub-tests. They might be tempted to configure the basic `terraform.Options` once, use `Clone()` to copy it for each sub-test, and within that subtest, assume they can override the `Vars` (for different settings) and `Logger` (to have different logging for each test)... But without a deep copy, the latter would be shared amongst all the tests...

Could we work around this by manually calling `copier.Copy` on `SshAgent` and `Logger`? Or do those have nested structs that would run into issues anyway? Alternatively, we can try to put a bigger warning on the `Clone` method docs about not sharing a cloned `Options` if they plan to override these specific fields...

Reply:
Pardon the delay, but addressed this in 2b1d1e5